ElementX/Sources/Services/Room/RoomProxy.swift

```diff
@@ -611,3 +619,17 @@ private class RoomTimelineListener: TimelineListener {
         onUpdateClosure(diff)
     }
 }
+
+private class UploadProgressListener: ProgressWatcher {
+    private let onUpdateClosure: (Double) -> Void
```

Comment:
Minor, and just thinking out loud here: maybe this would work instead of the dispatch queue?
```suggestion
    private let onUpdateClosure: @MainActor (Double) -> Void
```

Reply:
Maybe, definitely worth a shot 👍